code/go/0chain.net/smartcontract/storagesc/handler.go

```diff
@@ -2253,3 +2254,96 @@ func (srh StorageRestHandler) getBlobber(w http.ResponseWriter, r *http.Request)
 type readMarkersCount struct {
 	ReadMarkersCount int64 `json:"read_markers_count"`
 }
+
+/*getSearchHandler - Get result based on query*/
+// swagger:route GET /v1/screst/6dba10422e368813802877a85039d3985d96760ed844092319743fb3a76712d7/search search
+// Generic search endpoint
+//
+// parameters:
+//    + name: query
+//      description: Generic query string, supported inputs: Block hash, Round num, Transaction hash, File name, Content hash, Wallet address
+//      required: true
+//      in: query
+//      type: string
+//
+// responses:
+//  200:
+//  400:
+//  500:
+func (srh StorageRestHandler) getSearchHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		query = r.URL.Query().Get("query")
+	)
+
+	if len(query) == 0 {
+		common.Respond(w, r, nil, common.NewErrInternal("query param required"))
+		return
+	}
+
+	edb := srh.GetQueryStateContext().GetEventDB()
+	if edb == nil {
+		common.Respond(w, r, nil, common.NewErrInternal("no db connection"))
+		return
+	}
+
+	queryType, err := edb.GetGenericSearchType(query)
+	if err != nil {
+		common.Respond(w, r, nil, common.NewErrInternal(err.Error()))
+		return
+	}
+
+	switch queryType {
```

Comment:
there is no case 'FileName' here
which is there in gettingQueryType.. just to make sure we are covering all cases

Reply:
Good catch! I missed it. Added it.